Escape json

jbaiza opened the pull request:
Faced issues when tried to upgrade to latest plugin version (still used old SOAP version).
Added StringEscapeUtils.escapeJson in all JSON calls with possible user input.

bezda commented on src/main/java/info/bluefloyd/jenkins/RESTClient.java:
```diff
@@ -429,7 +450,7 @@ private RestResult doPost(URL url, String bodydata) throws IOException {
     result.setResultMessage(output.toString());
 
-    if ((conn.getResponseCode() == 200) || (conn.getResponseCode() == 201)) {
+    if ((conn.getResponseCode() == 200) || (conn.getResponseCode() == 201) || (conn.getResponseCode() == 204)) {
```
Seems like each operation is expecting different status code, It might be easier if we completly removed this validResult property and let caller methods check the responsCode

laszlomiklosik replied:
Using constants for the HTTP status codes would be another improvement idea

laszlomiklosik commented:
please fix the merge conflicts, I'll merge your change after that. thanks!

jbaiza commented:
I refactored request methods to remove unneeded duplications.
Moved response code checks to each method to check specific one.
Replaced constant status codes with class constants
